build.gradle

```diff
@@ -7,13 +7,19 @@ plugins {
 	id 'maven-publish'
 }
 
+group = 'com.iexec.blockchain'
+sourceCompatibility = '11'
+targetCompatibility = '11'
+
 ext {
 	openFeignVersion = '11.6'
+	gitBranch = 'git rev-parse --abbrev-ref HEAD'.execute().text.trim()
 }
 
-group = 'com.iexec.blockchain'
-sourceCompatibility = '11'
-targetCompatibility = '11'
+if (gitBranch != 'main' && gitBranch != 'master' && ! (gitBranch ==~ '(release|hotfix|support)/.*')) {
+	version            += '-NEXT-SNAPSHOT'
+	iexecCommonVersion += '-NEXT-SNAPSHOT'
```

Comment:
Why not handling iexec-common version manually?

Reply:
It is exactly my first suggestion in a previous comment.
`iexecCommonVersion` could stay outside of this versioning.
I just don't know if it keeps enough added value in this case.
Still, we can try it and see if it works.

Reply:
I also tend to prefer managing iexec-common version manually.

Reply:
done